File: drive-back/src/main/java/com/gabriel/drive_back/domain/Folder.java
package com.gabriel.drive_back.domain;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;

@Data
@Entity
@Table
@NoArgsConstructor
@AllArgsConstructor
public class Folder {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    private String name;
    private String createdTime;
    private int size;

    public Folder(String name) {
        this.name = name;
    }

    public Folder(String name, String createdTime) {
        this.name = name;
        this.createdTime = createdTime;
    }
}

File: drive-front/src/app/components/folder/Folder.ts
export class FolderClass {
  id: string;
  name: string;
  createdTime: string;
  size: string;
  isModalOpen: boolean;
  constructor(id: string, name: string, createdTime: string, size: string, isModalOpen: boolean) {
    this.isModalOpen = isModalOpen;
    this.size = size;
    this.createdTime = createdTime;
    this.id = id;
    this.name = name;
  }
}